grakn-core/src/main/java/ai/grakn/util/Schema.java

```diff
@@ -263,4 +263,14 @@ public TypeName getName() {
             return TypeName.of(name);
         }
     }
+
+    /**
+     *
+     * @param typeName The resource type name
+     * @param value The value of the resource
+     * @return A unique id for the resource
+     */
+    public static String generateResourceIndex(TypeName typeName, String value){
```

Comment:
I am starting to think we dumping too much in `Schema`. Maybe we should create a `CoreUtils` ? 

In my view `Schema` should only contain our hardcoded enums and configs.